NthKey/Model/DataManager.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  DataManager.swift
+//  DataManager
+//
+//  Created by Sergey Vinogradov on 22.03.2021.
+//  Copyright © 2021 Purple Dunes. Distributed under the MIT software
+//  license, see the accompanying file LICENSE.md
+
+import Foundation
+import CoreData
+import Combine
+
+/// Incapsulates all data changes and persistent store interaction inside
+final class DataManager: ObservableObject {
+    @Published var addressList: [AddressEntity] = []
+    @Published var walletList: [WalletEntity] = []
+
+    @Published var currentWallet: WalletEntity?
+
+    @Published var error: Error?
+
+    private let store: PersistentStore
+    fileprivate var cancellables = Set<AnyCancellable>()
+
+    private var dataWasPrepared = false
+    private var walletsRequest: NSFetchRequest<WalletEntity>
+
+    init(store: PersistentStore) {
+        self.store = store
+
+        walletsRequest = NSFetchRequest<WalletEntity>(entityName: "WalletEntity")
+        walletsRequest.sortDescriptors = [NSSortDescriptor(keyPath: \WalletEntity.label, ascending: true)]
+
+        setupObservables()
+    }
+
+    /// Load all needed values
+    func prepareData() {
+        guard !dataWasPrepared else { return }
+        dataWasPrepared = true
+
+        do {
+            walletList = try store.container.viewContext.fetch(walletsRequest)
+
+            /// Load previously selected wallet
+            if let value = UserDefaults.currentWalletDescriptor {
```

Comment:
Let's give each wallet an ID (or UUID). There may be (subtle) changes to the format of output descriptors in the future. Also I'd rather not store that info `UserDefaults` (maybe in the future we encrypt it in the database).

Reply:
fixed in e15a6d96739142925456e54c66423ddf718ac3bb